pkg/hub/managedclusterset/default_managedclusterset_label_controller.go

```diff
@@ -0,0 +1,95 @@
+package managedclusterset
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/openshift/library-go/pkg/controller/factory"
+	"github.com/openshift/library-go/pkg/operator/events"
+	"github.com/openshift/library-go/pkg/operator/resource/resourcemerge"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/klog/v2"
+	clientset "open-cluster-management.io/api/client/cluster/clientset/versioned"
+	clusterinformerv1 "open-cluster-management.io/api/client/cluster/informers/externalversions/cluster/v1"
+	clusterlisterv1 "open-cluster-management.io/api/client/cluster/listers/cluster/v1"
+	clusterv1 "open-cluster-management.io/api/cluster/v1"
+)
+
+const (
+	defaultManagedClusterSetValue = "default"
+)
+
+// defaultManagedClusterSetLabelController reconciles ManagedClusterSet label.
+type defaultManagedClusterSetLabelController struct {
+	clusterClient clientset.Interface
+	clusterLister clusterlisterv1.ManagedClusterLister
+	eventRecorder events.Recorder
+}
+
+// NewDefaultManagedClusterSetLabelController creates a new set ManagedClusterSet label controller
+func NewDefaultManagedClusterSetLabelController(
+	clusterClient clientset.Interface,
+	clusterInformer clusterinformerv1.ManagedClusterInformer,
+	recorder events.Recorder) factory.Controller {
+
+	c := &defaultManagedClusterSetLabelController{
+		clusterClient: clusterClient,
+		clusterLister: clusterInformer.Lister(),
+		eventRecorder: recorder.WithComponentSuffix("set-default-managed-cluster-set-label-controller"),
+	}
+
+	return factory.New().
+		WithInformersQueueKeyFunc(func(obj runtime.Object) string {
+			accessor, _ := meta.Accessor(obj)
+			return accessor.GetName()
+		}, clusterInformer.Informer()).
+		WithSync(c.sync).
+		ToController("DefaultManagedClusterSetLabelController", recorder)
+}
+
+func (c *defaultManagedClusterSetLabelController) sync(ctx context.Context, syncCtx factory.SyncContext) error {
+	managedClusterName := syncCtx.QueueKey()
+
+	klog.V(4).Infof("Reconciling ManagedClusterSetLabel of ManagedCluster %s", managedClusterName)
+
+	managedCluster, err := c.clusterLister.Get(managedClusterName)
+	if errors.IsNotFound(err) {
+		// Spoke cluster not found, could have been deleted, do nothing.
+		return nil
+	}
+	if err != nil {
+		return err
+	}
+
+	if !managedCluster.DeletionTimestamp.IsZero() {
+		// ManagedCluster is deleting, do nothing
+		return nil
+	}
+
+	if err := c.syncClusterSetLabel(ctx, managedCluster); err != nil {
+		return fmt.Errorf("failed to set default ManagedClusterSet label to ManagedCluster %s: %w", managedCluster.Name, err)
+	}
+	return nil
+}
+
+func (c *defaultManagedClusterSetLabelController) syncClusterSetLabel(ctx context.Context, managedCluster *clusterv1.ManagedCluster) error {
+	cluster := managedCluster.DeepCopy()
+	modified := false
+
+	clusterSetLabels := map[string]string{}
```

Comment:
Do not move all clusters into the default clusterset. Instead, please check if the cluster has already been a member of a clusterset first.

Reply:
done. Add if condition to check.